src/test/java/org/takes/facets/auth/social/PsLinkedinTest.java

```diff
@@ -24,32 +24,225 @@
+                            "linkedinTestUserInvalidFirstName.json"
+                        ).getPath()
+                    )
+                ).readObject()
+            );
+        MatcherAssert.assertThat(
+            new PsLinkedin(memberprofilejson)
+                .enter(REQUEST).next()
+                .properties().get("first_name"),
+            // @checkstyle MultipleStringLiteralsCheck (1 line)
+            Matchers.is("?")
+        );
+    }
+
+    /**
+     * Can parse invalid user last name field in json.
+     * @throws IOException If some problem inside
+     */
+    @Test
+    public void parsesInvalidLastNameField() throws IOException {
+        final MemberProfileJson memberprofilejson =
+            Mockito.mock(MemberProfileJson.class);
+        Mockito.when(
+            memberprofilejson.fetch(
+                Mockito.anyString(), Mockito.anyString(),
+                Mockito.any(Href.class)
+            )
+        ).thenReturn(
+                Json.createReader(
+                    new FileInputStream(
+                        PsLinkedinTest.class.getResource(
+                            "linkedinTestUserInvalidLastName.json"
+                        ).getPath()
+                    )
+                ).readObject()
+            );
+        MatcherAssert.assertThat(
+            new PsLinkedin(memberprofilejson)
+                .enter(REQUEST).next()
+                .properties().get("last_name"),
+            // @checkstyle MultipleStringLiteralsCheck (1 line)
+            Matchers.is("?")
+        );
+    }
+
+    /**
+     * Can parse invalid user ID field in json.
+     * @throws IOException If some problem inside
+     */
+    @Test(expected = ClassCastException.class)
+    public void parsesInvalidIdField() throws IOException {
+        final MemberProfileJson memberprofilejson =
+            Mockito.mock(MemberProfileJson.class);
+        Mockito.when(
+            memberprofilejson.fetch(
+                Mockito.anyString(), Mockito.anyString(),
+                Mockito.any(Href.class)
+            )
+        ).thenReturn(
+                Json.createReader(
+                    new FileInputStream(
+                        PsLinkedinTest.class.getResource(
+                            "linkedinTestUserInvalidId.json"
+                        ).getPath()
+                    )
+                ).readObject()
+            );
+        try {
+            new PsLinkedin(memberprofilejson)
+                .enter(REQUEST);
+        } catch (final ClassCastException ex) {
+            MatcherAssert.assertThat(
+                ex.getMessage(),
+                Matchers.containsString("cannot be cast to")
+            );
+            throw ex;
+        }
+    }
+
+    /**
+     * Can exit.
+     * @throws IOException If some problem inside
+     */
+    @Test
+    public void exitsSuccessfully() throws IOException {
+        final MemberProfileJson memberprofilejson =
```

Comment:
@ikhvostenkov again `json`, maybe this part with mocking `MemberProfileJson` and adding `when` to it could be moved into single private method that would accept just the filename as a parameter.